File: libs/domain-models/registry.ts
export interface Team {
  id: string;
  name: string;
  memberCount: number;
  serviceCount: number;
}

export interface User {
  id: string;
  email: string;
  roles: string[];
  teamIds: string[];
}

export interface ServiceOwnership {
  serviceId: string;
  teamId: string;
  ownerUserId: string;
}

File: libs/domain-models/registry.py
from dataclasses import dataclass


@dataclass(slots=True)
class Team:
    id: str
    name: str
    member_count: int
    service_count: int


@dataclass(slots=True)
class User:
    id: str
    email: str
    roles: list[str]
    team_ids: list[str]


@dataclass(slots=True)
class ServiceOwnership:
    service_id: str
    team_id: str
    owner_user_id: str
